File: app/src/main/java/com/unistay/android/domain/model/model/ViewHistory.kt
package com.unistay.android.domain.model.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "view_history")
data class ViewHistory(
    @PrimaryKey(autoGenerate = true) val historyId: Int = 0,
    val userId: String,
    val roomId: String,
    val viewedAt: Long = System.currentTimeMillis()
)

File: UniStay.Api/Models/ViewHistory.cs
using System.ComponentModel.DataAnnotations;

namespace UniStay.Api.Models
{
    public class ViewHistory
    {
        [Key]
        public int HistoryId { get; set; }
        public string UserId { get; set; } = string.Empty;
        public string RoomId { get; set; } = string.Empty;
        public long ViewedAt { get; set; } = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
    }
}
